signxml/xades/__init__.py

```diff
@@ -339,8 +341,7 @@ def _generate_xades_ssp_elements(self, options_struct):
                 defined in IETF RFC 5035 [17].
                 """
                 serial_element = XADES132.IssuerSerialV2(
-                    DS.X09IssuerName(issuer_string(cert)),
-                    DS.X509SerialNumber(f"cert.get_serial_number()")
+                    b64encode(_der.encode_der_integer(cert.get_serial_number()))
```

Comment:
Well, that's not the ASN object required here, it's the IssuerSerial oject...

Reply:
I read this:
https://www.pyopenssl.org/en/stable/api/crypto.html#OpenSSL.crypto.X509.get_serial_number
I am calling `get_serial_number()`
Certificate has:
* issuer
* issuer serial number

Reply:
I see:

|   |   |
| --- | ---- |
| Returns: | The serial number. |
| Return type: | int |

But waht needs der encoding is this RFC `IssuerSerial` asn object

Reply:
```
        IssuerSerial ::= SEQUENCE {
            issuer                   GeneralNames,
            serialNumber             CertificateSerialNumber
       }
```
taken from: https://tools.ietf.org/html/rfc5035

Reply:
`GeneralNames` can be quite some different names, but I haven't yet analyzed how that poses encoding challanges...

Reply:
Please note also: it's a sequence, or array (if I understand the ASN spec grammar correctly).

Reply:
IMO `DER` representation must be done at application (signxml) level; so `pyopenssl` and `cryptography` provide `API` to solve this. AFAIK i do it.

`IssueSerial` comes from certificate from `pyopenssl` in 2 objects

* `SerialName`
* `SerialNumber`

Here we need to identify what needs the `V2` on new spec; in legacy they has 2 elements on XML and i understood in V2 just one: `SerialNumber`

You can check my PR X509Name class and my own implementation to reverse `get_components()`

In terms of X509 Object all Issue serial is represented by:

![imagen](https://user-images.githubusercontent.com/31282/71326295-78f61200-24be-11ea-822d-60940ab8bfa9.png)
 
as RFC mentioned defined.

Reply:
Why not construe the mentioned ASN object using some ASN generator? (`pyasn.1` - I guess) - and then DER encode through one of the APIs you mention.

Or could you point me to the pyopenssl IssuerSerial object I haven't seen it. (is it an sequence?)

Reply:
As for the name string, what's the difference between your implementation and https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Name.rfc4514_string ?